File: fyc-api-rest/src/main/java/fr/esgi/fyc_api_rest/mapper/SeanceMapper.java
package fr.esgi.fyc_api_rest.mapper;

import fr.esgi.fyc_api_rest.business.Seance;
import fr.esgi.fyc_api_rest.dto.request.SeanceRequestDTO;
import fr.esgi.fyc_api_rest.dto.response.SeanceResponseDTO;
import fr.esgi.fyc_api_rest.service.SportifService;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingConstants;
import org.mapstruct.ReportingPolicy;

@Mapper(
        unmappedTargetPolicy = ReportingPolicy.IGNORE,
        componentModel = MappingConstants.ComponentModel.SPRING,
        uses = {SportifService.class, ExerciceMapper.class}
)
public interface SeanceMapper {
    @Mapping(target = "sportif", source = "idSportif")
    public Seance toEntity(SeanceRequestDTO seanceDTO);

    @Mapping(target = "sportifResponse", source = "sportif")
    @Mapping(target = "exercicesResponse", source = "exercices")
    public SeanceResponseDTO toResponseDTO(Seance seance);
}

File: fyc-rest-api-csharp-test/CometeAPI/Mapper/SeanceMapper.cs
using AutoMapper;
using FycRestApi.Business;
using FycRestApi.DTO.Request;
using FycRestApi.DTO.Response;
using FycRestApi.Service.Interface;

namespace FycRestApi.Mapper;

public class SeanceMapper
{
    private readonly IMapper _mapper;
    private readonly ISportifService _sportifService;

    public SeanceMapper(IMapper mapper, ISportifService sportifService)
    {
        this._mapper = mapper;
        this._sportifService = sportifService;
    }

    public async Task<Seance> toEntity(SeanceRequestDTO requestDTO)
    {
        Seance seance = _mapper.Map<Seance>(requestDTO);
        seance.sportif = await _sportifService.findById(requestDTO.idSportif);
        return seance;
    }

    public SeanceResponseDTO toDTO(Seance entity)
    {
        SeanceResponseDTO seanceResponseDTO = _mapper.Map<SeanceResponseDTO>(entity);
        seanceResponseDTO.sportifResponse = _mapper.Map<SportifResponseDTO>(entity.sportif);
        return seanceResponseDTO;
    }
}